packages/shared/src/components/modals/ReportPostModal.tsx

```diff
@@ -0,0 +1,84 @@
+import React, { ReactElement, MouseEvent, useState, useRef } from 'react';
+import { Radio } from '../fields/Radio';
+import { Post } from '../../graphql/posts';
+import { ModalCloseButton } from './ModalCloseButton';
+import { Checkbox } from '../fields/Checkbox';
+import { ModalProps } from './StyledModal';
+import { Button } from '../buttons/Button';
+import {
+  ReportModal,
+  ReportHeading,
+  ReportDescription,
+  ReportButtonsCenter,
+} from './ReportModal';
+
+export interface Props extends ModalProps {
+  postIndex: number;
+  post: Post;
+  onReport: (postIndex, postId, reason, comment, blockSource) => unknown;
+}
+
+const reportReasons: { value: string; label: string }[] = [
+  { value: 'BROKEN', label: 'Broken link' },
+  { value: 'CLICKBAIT', label: 'Clickbait' },
+  { value: 'LOW', label: 'Low-quality content' },
+  { value: 'NSFW', label: 'NSFW' },
+  { value: 'OTHER', label: 'Other' },
+];
+
+export default function RepostPostModal({
+  postIndex,
+  post,
+  onReport,
+  ...props
+}: Props): ReactElement {
+  const [reason, setReason] = useState(null);
+  const [comment, setComment] = useState<string>();
+  const inputRef = useRef<HTMLInputElement>();
+
+  const onReportPost = async (event: MouseEvent): Promise<void> => {
+    onReport(postIndex, post, reason, comment, inputRef.current?.checked);
+    props.onRequestClose(event);
+  };
+
+  return (
+    <ReportModal {...props}>
+      <ModalCloseButton onClick={props.onRequestClose} />
+      <ReportHeading>Report article</ReportHeading>
+      <ReportDescription>&quot;{post?.title}&quot;</ReportDescription>
+      <div className="w-full">
+        <Radio
+          className="px-2 mt-2 mb-4"
+          name="report_reason"
+          options={reportReasons}
+          value={reason}
+          onChange={setReason}
+        />
+        <div className="-mx-8 h-px bg-theme-divider-secondary" />
+        <p className="px-2 mt-6 mb-1 font-bold typo-caption1">
+          Anything else you&apos;d like to add?
+        </p>
+        <textarea
+          onChange={(event) => setComment(event.target.value)}
+          className="self-stretch p-2 mb-4 w-full h-20 bg-theme-float rounded-10 resize-none typo-body"
+        />
+        <Checkbox
+          ref={inputRef}
+          name="blockSource"
+          className="self-center px-1 mx-2 mb-6 font-normal"
```

Comment:
The padding shouldn't be necessary for the current setup since we already have the margin for the missing `0.5rem` 
```suggestion
          className="self-center mx-2 mb-6 font-normal"
```

Reply:
Added separate topic as this will affect the other implementation as well:
https://dailydotdev.atlassian.net/jira/software/projects/DD/boards/1?selectedIssue=DD-291